torchgeo/datamodules/cdlsentinel2.py

```diff
@@ -0,0 +1,107 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""CDLSentinel2 datamodule."""
+
+from typing import Any, Optional, Union
+
+import kornia.augmentation as K
+import torch
+from kornia.constants import DataKey, Resample
+
+from ..datasets import CDL, Sentinel2, random_grid_cell_assignment
+from ..samplers import GridGeoSampler, RandomBatchGeoSampler
+from ..samplers.utils import _to_tuple
+from ..transforms import AugmentationSequential
+from .geo import GeoDataModule
+
+
+class CDLSentinel2DataModule(GeoDataModule):
+    """LightningDataModule implementation for the CDL dataset.
+
+    .. versionadded:: 0.6
+    """
+
+    def __init__(
+        self,
+        batch_size: int = 64,
+        patch_size: Union[int, tuple[int, int]] = 16,
+        length: Optional[int] = None,
+        num_workers: int = 0,
+        **kwargs: Any,
+    ) -> None:
+        """Initialize a new CDLSentinel2DataModule instance.
+
+        Args:
+            batch_size: Size of each mini-batch.
+            patch_size: Size of each patch, either ``size`` or ``(height, width)``.
+            length: Length of each training epoch.
+            num_workers: Number of workers for parallel data loading.
+            **kwargs: Additional keyword arguments passed to
+                :class:`~torchgeo.datasets.CDL`.
+        """
+        cdl_signature = "cdl_"
+        sentinel2_signature = "sentinel2_"
+        self.cdl_kwargs = {}
+        self.sentinel2_kwargs = {}
+        for key, val in kwargs.items():
```

Comment:
Can you add some comments to describe what this logic is doing?

Reply:
We have the same logic in all of our A & B GeoDataModules. We could extract it out into a helper function now that it's becoming common. Would that help with the documentation?